### PR TITLE
[TT-4028] Fix race condition in httpdashboardhandler

### PR DESCRIPTION
**Pull request by titpetric**
## Description

Dashboardhandler has a race condition, PR fixes it using atomic operations. It tentatively fixes a bug where StopBeating would stop the beat loop, but would immediately reset it to started. Moved the "STARTED" operation higher up.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Refactoring or add test (improvements in base code or adds test coverage to functionality)

## Checklist

- [ ] I ensured that the documentation is up to date
- [ ] I explained why this PR updates go.mod in detail with reasoning why it's required
- [x] I would like a code coverage CI quality gate exception and have explained why

### CODE REVIEW

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4844/merge
Commit:  
Triggered by: pull_request (@titpetric)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/4372621204)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4844/merge
Commit: 70c7539ccb563535827ba322c82b2b6d57df0286 
Triggered by: pull_request (@jeffy-mathew)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/4530172259)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4844/merge
Commit: d28ea78ca57324df2f631a7ba6f7e821cfaf4997 
Triggered by: pull_request (@jeffy-mathew)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/4530475010)

**Review by jeffy-mathew — APPROVED** [review]

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4844/merge
Commit: bbd46bda8cced4284df993cae62fc78c84b4266c 
Triggered by: pull_request (@jeffy-mathew)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/4552802165)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4844/merge
Commit: f9604ff01fc2067d1dd60ac51bb5ef36ed8b8d5b 
Triggered by: pull_request (@jeffy-mathew)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/4553413678)

File: gateway/dashboard_register.go [hunk 1]
```diff
@@ -30,6 +31,18 @@ type DashboardServiceSender interface {
 	NotifyDashboardOfEvent(interface{}) error
 }
 
+// Constants for heartBeatStopSentinel indicators.
+//
+// Go 1.17 adds atomic.Value.Swap which is great, but 1.19
+// adds atomic.Bool and other types. This is a go <1.13 cludge.
+const (
+	// Zero value - the handlers started
+	STARTED = 0
```
Comment (furkansenharputlu) [inline gateway/dashboard_register.go:40]: - `HeartbeatStarted`
- `HeartbeatStopped`

File: gateway/dashboard_register.go [hunk 2]
```diff
@@ -213,26 +227,30 @@ func (h *HTTPDashboardHandler) Ping() error {
 		h.Gw.initialiseClient())
 }
 
+func (h *HTTPDashboardHandler) isStopped() bool {
```
Comment (furkansenharputlu) [inline gateway/dashboard_register.go:230]: `isHeartBeatStopped`

**Review by furkansenharputlu — APPROVED** [review]

**Comment by Tyk-ITS** [comment]
**API tests result: failure** :no_entry_sign: 
Branch used: refs/pull/4844/merge
Commit: c66f0ffd7b44c35197bcf8a14601464c9b276fa5 
Triggered by: pull_request (@jeffy-mathew)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/4563605284)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4844/merge
Commit: 536737c6c3419939722c6d1dce905c79f8070ceb 
Triggered by: pull_request (@jeffy-mathew)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/4572679019)

**Pull request merged**